src/command_modules/azure-cli-role/azure/cli/command_modules/role/_params.py

```diff
@@ -42,6 +42,11 @@ def load_arguments(self, _):
                    help="resource scopes and roles the application requires access to. Should be in manifest json format. See examples below for details")
         c.argument('native_app', arg_type=get_three_state_flag(), help="an application which can be installed on a user's device or computer")
 
+    with self.argument_context('ad app grant') as c:
+        c.argument('identifier', options_list=['--id'], help='clientId of the app you want to grant permissions to')
```

Comment:
I suggest do not redefine, rather let the [default one](https://github.com/Azure/azure-cli/blob/dev/src/command_modules/azure-cli-role/azure/cli/command_modules/role/_params.py#L29) take over

Reply:
Implemented